src/Engine/Waypoint/Waypoints.cpp

```diff
@@ -420,4 +420,6 @@ Waypoints::AddGotoPoint(const GeoPoint& location,
   Append(std::move(new_goto_point));
 
   Optimise();
+  WaypointPtr new_goto_point_ptr = LookupName(_T("(goto)"));
```

Comment:
@lordfolken How can I pass the `new_goto_point` without performing another lookup?
Sorry about the question, but this is my very first effort in C++